tests/components/rest/test_sensor.py

```diff
@@ -1,696 +1,665 @@
+    respx.get(
+        "http://localhost",
+        status_code=200,
+        headers={"content-type": "text/xml"},
+        content="<dog>abc</dog>",
+    )
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "value_template": "{{ value_json.dog }}",
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+    state = hass.states.get("sensor.foo")
+    assert state.state == "abc"
+    assert state.attributes[ATTR_UNIT_OF_MEASUREMENT] == DATA_MEGABYTES
+
+
+@respx.mock
+async def test_update_with_json_attrs(hass):
+    """Test attributes get extracted from a JSON result."""
+
+    respx.get(
+        "http://localhost",
+        status_code=200,
+        headers={"content-type": CONTENT_TYPE_JSON},
+        content='{ "key": "some_json_value" }',
+    )
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "value_template": "{{ value_json.key }}",
+                "json_attributes": ["key"],
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+    state = hass.states.get("sensor.foo")
+    assert state.state == "some_json_value"
+    assert state.attributes["key"] == "some_json_value"
+
+
+@respx.mock
+async def test_update_with_no_template(hass):
+    """Test update when there is no value template."""
+
+    respx.get(
+        "http://localhost",
+        status_code=200,
+        headers={"content-type": CONTENT_TYPE_JSON},
+        content='{ "key": "some_json_value" }',
+    )
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+                "json_attributes": ["key"],
+                "name": "foo",
+                "unit_of_measurement": DATA_MEGABYTES,
+                "verify_ssl": "true",
+                "timeout": 30,
+                "headers": {"Accept": "text/xml"},
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
+
+    state = hass.states.get("sensor.foo")
+    assert state.state == '{ "key": "some_json_value" }'
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)